runtime/interpreter/value.go

```diff
@@ -73,7 +73,7 @@ type ConcatenatableValue interface {
 
 type EquatableValue interface {
 	Value
-	Equal(other Value) BoolValue
+	Equal(interpreter *Interpreter, other Value) BoolValue
```

Comment:
The `interpreter` is only useful for `TypeValue`, and for other types, it's not needed.

Another approach would be to ask if the type of the values need to be converted first. If yes, then convert them using the interpreter, and then call equal with the converted values. So if a value is a TypeValue, its type needs to be converted to `sema.Type`, and `sema.Type` has `Equal` defined.

On a side note, we might need to prevent `convertStaticToSemaType` from returning a `TypeValue`. Because otherwise, the `Equal` will trigger another `convertStaticToSemaType` call, and potentially run into endless loop.

Reply:
Yeah, maybe conditionally converting might be a good idea, I'll look into it

Reply:
As I noted above, the conversion is always performed so that the equality functions of the `sema.Type` hierarchy can be used, and the equality functions don't have to be reimplemented for the `interpreter.StaticType` hierarchy.

I'm not quite following the last paragraph: `convertStaticToSemaType` doesn't return a `TypeValue`, it returns a `sema.Type`.